ReactAndroid/src/main/java/com/facebook/react/modules/network/OkHttpClientProvider.java

```diff
@@ -43,18 +59,51 @@ public static OkHttpClient getOkHttpClient() {
     }
     return sClient;
   }
+
+  private static class EvictIdleConnectionsTask extends AsyncTask {
+    @Override
+    protected Object doInBackground(Object[] objects) {
+      for (OkHttpClient client: sClients) {
+        client.connectionPool().evictAll();
+      }
+      return null;
+    }
+  }
+
+  public static void addNetworkListenerToEvictClientConnectionsOnDisconnect(Context context) {
+    BroadcastReceiver br = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        Bundle extras = intent.getExtras();
+        NetworkInfo info = extras.getParcelable("networkInfo");
+        NetworkInfo.State state = info.getState();
+        if (state == NetworkInfo.State.DISCONNECTED) {
+          new EvictIdleConnectionsTask().execute();
```

Comment:
How did you choose this approach, with an AsyncTask, for executing this work?

I know this is a pretty tricky area of Android programming, and I don't really know what the right way to do it is; so I'm curious what docs you relied on (if you found some that helped) or how else you chose this version.

Reading docs a bit (both the method's reference doc I linked above, and the [guide](https://developer.android.com/guide/components/broadcasts), which tends to be a very helpful genre of Android documentation when it exists):
* this method is called on the main thread, so we shouldn't do anything slow;
* I can't tell from OkHttp's docs if this method might be slow -- seems like probably it shouldn't but I'm not sure;
* there's a warning about async tasks, saying the process might get killed after the receiver returns. With a recommendation to use `goAsync` to avoid that... whose own doc says it was introduced in API level 11, so plenty ancient enough for RN.

So I think probably the async isn't necessary, but it may be wise to do it anyway to be sure; but if we do, we should probably use `goAsync`.

Reply:
If `connectionPool().evictAll()` is called on the UI thread, a NetworkOnMainThreadException is thrown. Android wants you to execute any networking operations in the background. I chose AsyncTask because the internet told me to (some high-voted SO answer used it and another blog post for which I cannot find the link anymore). Looking a bit more into this, it seems like AsyncTask -can potentially cause trouble if it's performing a longer operation, because it might get terminated when the activity changes (e.g. screen rotation). That shouldn't be a problem because evictAll is fast, but `goAsync` seems definitely better since it's built for BroadcastReceivers.

Reply:
(Replied in main thread.)